amazon-chime-sdk-js/src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -574,6 +585,251 @@ export default class DefaultTransceiverController
+        }
+        controller.enqueue(encodedFrame);
+      }
+
+      let rcount = 0;
+      function decodeFunction(encodedFrame: any, controller: any) {
+        if (rcount++ < 30) {
+          // dump the first 30 packets
+          dump(encodedFrame, 'recv');
+        }
+        const view = new DataView(encodedFrame.data);
+        const checksum =
+          encodedFrame.data.byteLength > 4
+            ? view.getUint32(encodedFrame.data.byteLength - 4)
+            : false;
+        if (currentCryptoKey) {
+          if (checksum !== 0xdeadbeef) {
+            console.log('Corrupted frame received, checksum ' + checksum.toString(16));
+            return; // This can happen when the key is set and there is an unencrypted frame in-flight.
+          }
+          const keyIdentifier = view.getUint8(encodedFrame.data.byteLength - 5);
+          if (keyIdentifier !== currentKeyIdentifier) {
+            console.log(
+              `Key identifier mismatch, got ${keyIdentifier} expected ${currentKeyIdentifier}.`
+            );
+            return;
+          }
+
+          const newData = new ArrayBuffer(encodedFrame.data.byteLength - 5);
+          const newView = new DataView(newData);
+          const cryptoOffset = useCryptoOffset ? frameTypeToCryptoOffset[encodedFrame.type as keyof typeof frameTypeToCryptoOffset] : 0;
+
+          for (let i = 0; i < cryptoOffset; ++i) {
+            newView.setInt8(i, view.getInt8(i));
+          }
+          for (let i = cryptoOffset; i < encodedFrame.data.byteLength - 5; ++i) {
+            const keyByte = currentCryptoKey.charCodeAt(i % currentCryptoKey.length);
+            newView.setInt8(i, view.getInt8(i) ^ keyByte);
+          }
+          encodedFrame.data = newData;
+        } else if (checksum === 0xdeadbeef) {
+          return; // encrypted in-flight frame but we already forgot about the key.
+        }
+        controller.enqueue(encodedFrame);
+      }
+
+      function handleTransform(operation: any, readable: any, writable: any) {
+        if (operation === 'encode') {
+          const transformStream = new TransformStream({
+            transform: encodeFunction,
+          });
+          readable.pipeThrough(transformStream).pipeTo(writable);
+        } else if (operation === 'decode') {
+          const transformStream = new TransformStream({
+            transform: decodeFunction,
+          });
+          readable.pipeThrough(transformStream).pipeTo(writable);
+        }
+      }
+
+      if (event.data.operation === 'encode' || event.data.operation === 'decode') {
+        return handleTransform(event.data.operation, event.data.readable, event.data.writable);
+      }
+      if (event.data.operation === 'setCryptoKey') {
+        if (event.data.currentCryptoKey !== currentCryptoKey) {
+          currentKeyIdentifier++;
+        }
+        currentCryptoKey = event.data.currentCryptoKey;
+        useCryptoOffset = event.data.useCryptoOffset;
+      }
```

Comment:
All of these should go to the worker code. To make sure it is working properly, make sure the log statements of dump function are printed.